docs/user-guides/scripts.md

```diff
@@ -0,0 +1,423 @@
+us to simplify writing scripts with parameters and artifacts that require additional fields such as a `description` or
+alternative `name`.
+
+This feature must be enabled by setting the `experimental_feature` flag `script_annotations` on the global config.
+
+```py
+global_config.experimental_features["script_annotations"] = True
+```
+
+### Parameters
+
+In Hera, we can currently specify inputs inside the `@script` decorator as follows:
+
+```python
+@script(
+    inputs=[
+        Parameter(name="an_int", description="an_int parameter", default=1, enum=[1, 2, 3]), 
+        Parameter(name="a_bool", description="a_bool parameter", default=True, enum=[True, False]), 
+        Parameter(name="a_string", description="a_string parameter", default="a", enum=["a", "b", "c"])
+    ]
+)
+def echo_all(an_int=1, a_bool=True, a_string="a"):
+    print(an_int)
+    print(a_bool)
+    print(a_string)
+```
+
+Notice how the `name` and `default` values are duplicated for each `Parameter`. Using annotations, we can rewrite this
+as:
+
+```python
+@script()
+def echo_all(
+    an_int: Annotated[int, Parameter(description="an_int parameter", default=1, enum=[1, 2, 3])], 
+    a_bool: Annotated[bool, Parameter(description="a_bool parameter", default=True, enum=[True, False])], 
+    a_string: Annotated[str, Parameter(description="a_string parameter", default="a", enum=["a", "b", "c"])]
+):
+    print(an_int)
+    print(a_bool)
+    print(a_string)
+```
+
+The fields allowed in the `Parameter` annotations are: `name`, `default`, `enum`, and `description`.
+
+### Artifacts
+
+> Note: Artifact annotations are only supported when used with the `RunnerScriptConstructor`.
+
+
+The feature is even more powerful for `Artifact`s. In Hera we are currently able to specify `Artifact`s in `inputs`, but
+the given path is not programmatically linked to the code within the function unless defined outside the scope of the
+function:
+
+```python
+@script(inputs=Artifact(name="my-artifact", path="/tmp/file"))
+def read_artifact():
+    with open("/tmp/file") as a_file:  # Repeating "/tmp/file" is prone to human error!
+        print(a_file.read())
+
+# or
+
+MY_PATH = "/tmp/file"  # Now accessible outside of the function scope!
+@script(inputs=Artifact(name="my-artifact", path=MY_PATH))
+def read_artifact():
+    with open(MY_PATH) as a_file:
+        print(a_file.read())
+```
+
+By using annotations we can avoid repeating the `path` of the file, and the function can use the variable directly as a
+`Path` object, with its value already set to the given path:
+
+```python
+@script(constructor="runner")
+def read_artifact(an_artifact: Annotated[Path, Artifact(name="my-artifact", path="/tmp/file")]):
+    print(an_artifact.read_text())
+```
+
+The fields allowed in the `Artifact` annotations are: `name`, `path`, and `loader`.
+
+### Artifact Loaders
+
+In case you want to load an object directly from the Artifact path, we allow two types of loaders besides the default
```

Comment:
```suggestion
In case you want to load an object directly from the `Artifact` path, we allow two types of loaders besides the default
```

Reply:
Gone for
```suggestion
In case you want to load an object directly from the `path` of the `Artifact`, we allow two types of loaders besides the default
```